Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -1511,20 +1518,20 @@ bool remove_self_intersections_one_step(TriangleMesh& tm,
     }
     else
       /// \todo check whether this is more expensive than the previous code above
-      BOOST_FOREACH(face_descriptor f, faces_to_remove)
+      for(face_descriptor f : faces_to_remove)
         Euler::remove_face(halfedge(f, tm), tm);
 
     if (verbose)
       std::cout << "  DEBUG: " << faces_to_remove.size() << " triangles removed" << std::endl;
 
     /// now get one halfedge per hole
     std::set<halfedge_descriptor> visited;
-    BOOST_FOREACH(halfedge_descriptor h, boundary_hedges)
+    for(halfedge_descriptor h : boundary_hedges)
     {
       if (visited.insert(h).second)
       {
         one_halfedge_per_border.push_back(h);
-        BOOST_FOREACH(halfedge_descriptor hh, halfedges_around_face(h, tm))
```

Comment:
That `BOOST_FOREACH` is nested... Would it be sufficient to convert only that one?